### PR TITLE
Replace QueryItem with a GAT

### PR DESCRIPTION
**Pull request by Ralith**
Decent cleanup and clarity improvement. Waiting on 1.65 release.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ repository = "https://github.com/Ralith/hecs"
 readme = "README.md"
 keywords = ["ecs", "entity"]
 categories = ["data-structures", "game-engines", "no-std"]
-rust-version = "1.57"
+rust-version = "1.66"
```
Comment (a1phyr) [inline Cargo.toml:12]:
```suggestion
rust-version = "1.65"
```

GAT will be stabilized even sooner !

Comment (Ralith) [reply Cargo.toml:12]: Good catch, thanks!

**Comment by casimcdaniels** [comment]
@Ralith 1.65 was released back in November. Should this be merged now?

**Review by casimcdaniels — APPROVED** [review]

**Comment by Ralith** [comment]
Might as well!

**Comment by casimcdaniels** [comment]
cheers!

**Pull request merged**